pkg/dtls/heartbeat.go

```diff
@@ -2,12 +2,14 @@ package dtls
 
 import (
 	"bytes"
+	"errors"
 	"net"
 	"sync/atomic"
 	"time"
 )
 
 var maxMessageSize = 65535
```

Comment:
Since this is internal and we control  this and the place this is interacting with can we make this buffers the same size (32kB --- `32*1024`) to match the buffer that is going to be reading from it? If that is going to mess with message sizes when sending SCTP with heartbeats then it might be fine as is.

Reply:
We could, yeah. I am wondering if it would be better to change the buffer in `halfPipe` to 65535 so that we can have a larger maximum message? The reason I put 65535 is because that's the pion SCTP's [maxMessageSize](https://github.com/pion/sctp/blob/c08b7f4b71d950fe91fd8cdd34c43e6e27592b28/association.go#L55), but we can [change](https://pkg.go.dev/github.com/pion/sctp#Association.SetMaxMessageSize) that if we want as well.

Reply:
Actually I think I got `halfPipe` confused with the SCTP code. It is able to read in messages larger than 32kB because it will just read the next part in the for loop. SCTP had a maxMessageSize because it is a message-oriented stream instead of the byte-stream that `halfPipe` expects, which meant that it must read each message in full. Thanks to the additions in #253, it should now behave like a byte-stream and work well with `halfPipe` on large messages.